### PR TITLE
[router-table] add new helper methods `FindNextHopOf()` and renames

### PR DESCRIPTION
**Pull request by abtink**
This PR contains two related commits:

**[router-table] add new helper methods `FindNextHopOf()` and renames**

This commit contains smaller enhancements in `RouterTable` class:

- New method `FindNextHopOf()` is added to find the next hop of a
  given router.
- Methods which find router are renamed: `FindRouterById()` to search
  for a given Router ID, `FindRouterByRloc16()` to search for a given
  RLOC16.
- The methods which find a neighbor router are declared as `private`
  and only used by `friend` class `NeighborTable`. Other modules use
  `NeighborTable` methods to find neighbors.

**[router-table] move `FillRouteTlv()` to `RouterTable`**
    
This commit moves `FillRouteTlv()` method to `RouterTable` class.

### CODE REVIEW

File: src/core/thread/router_table.cpp [hunk 1]
```diff
@@ -368,17 +368,17 @@ Router *RouterTable::GetNeighbor(uint16_t aRloc16)
...
 }
 
-Router *RouterTable::GetNeighbor(const Mac::Address &aMacAddress)
+Router *RouterTable::FindNeighbor(const Mac::Address &aMacAddress)
 {
     return FindRouter(Router::AddressMatcher(aMacAddress, Router::kInStateValid));
 }
 
-const Router *RouterTable::GetRouter(uint8_t aRouterId) const
+const Router *RouterTable::FindRouterForId(uint8_t aRouterId) const
```
Comment (jwhui) [inline src/core/thread/router_table.cpp:381]: nit: `FindRouterById` seems more natural to me. Thoughts?

File: src/core/thread/router_table.cpp [hunk 2]
```diff
@@ -393,7 +393,17 @@ const Router *RouterTable::GetRouter(uint8_t aRouterId) const
     return router;
 }
 
-Router *RouterTable::GetRouter(const Mac::ExtAddress &aExtAddress)
+const Router *RouterTable::FindRouterForRloc16(uint16_t aRloc16) const
```
Comment (jwhui) [inline src/core/thread/router_table.cpp:396]: nit: `FindRouterByRloc16` seems more natural to me. Thoughts?

File: src/core/thread/router_table.cpp (hunk 1 above)
Comment (abtink) [reply src/core/thread/router_table.cpp:381]: SGTM. I actually used `FindRouterById()` in the commit message (when describing the change) but in the code used the `FindRouterForId()`. 😄

Comment (abtink) [reply src/core/thread/router_table.cpp]: Updated in new push.

**Review by jwhui — APPROVED** [review]

**Pull request merged**